python/shotgun_model/shotgun_query_model.py

```diff
@@ -170,10 +170,14 @@ def __init__(self, parent, bg_load_thumbs, bg_task_manager=None):
     ############################################################################
     # public methods
 
-    def clear(self):
+    def clear(self, remove_data_handler=True):
```

Comment:
This method shadows the built-in http://doc.qt.io/archives/qt-4.8/qstandarditemmodel.html#clear - i am not sure it makes sense to extend these qt model interface methods with our own parameters.

Reply:
I've refactored the fix here to no longer change the class's public interface. You're absolutely right the we should try to avoid that wherever possible!